spec/factories/course_condition_achievements.rb

```diff
@@ -3,5 +3,6 @@
           class: Course::Condition::Achievement.name, aliases: [:achievement_condition] do
     course
     achievement
+    association :conditional, factory: :course_achievement
```

Comment:
Not entirely sure, but the specs before weren't using this association so it went unnoticed.